lib/inertia_rails/controller.rb

```diff
@@ -1,37 +1,32 @@
 require_relative "inertia_rails"
+require_relative "helper"
 
 module InertiaRails
   module Controller
     extend ActiveSupport::Concern
 
     included do
-      helper_method :inertia_headers
-
-      before_action do
-        error_sharing = proc do
-          # :inertia_errors are deleted from the session by the middleware
-          if @_request && session[:inertia_errors].present?
-            { errors: session[:inertia_errors] }
-          else
-            {}
-          end
-        end
-
-        @_inertia_shared_plain_data ||= {}
-        @_inertia_shared_blocks ||= [error_sharing]
-        @_inertia_html_headers ||= []
-      end
+      helper ::InertiaRails::Helper
 
       after_action do
-        cookies['XSRF-TOKEN'] = form_authenticity_token unless !protect_against_forgery?
+        cookies['XSRF-TOKEN'] = form_authenticity_token if protect_against_forgery?
       end
     end
 
     module ClassMethods
-      def inertia_share(hash = nil, &block)
-        before_action do
-          @_inertia_shared_plain_data = @_inertia_shared_plain_data.merge(hash) if hash
-          @_inertia_shared_blocks = @_inertia_shared_blocks + [block] if block_given?
+      def inertia_share(**attrs, &block)
```

Comment:
This is a breaking change, should we continue supporting hashes or at least throw a deprecation at first, wdyt?

```ruby
inertia_share({foo: 'bar'}) # => throws error
```